spras/containers.py

```diff
@@ -388,7 +388,7 @@ def prepare_volume(filename: Union[str, PurePath], volume_base: Union[str, PureP
     if isinstance(filename, PurePath):
         filename = str(filename)
 
-    filename_hash = hash_filename(filename, config.config.hash_length)
+    filename_hash = hash_filename(filename, config.config.container_settings.hash_length)
```

Comment:
It isn't intuitive to me to look for the `hash_length` in the `container settings`.

Reply:
(See comment from the above review)

Reply:
Looking ahead at #390 I'm still having a hard time understanding this design. The hash length is a global config property, not a container setting. Now `prepare_volume`'s signature will have to be modified so that a `ProcessedContainerSettings` instance has to be supplied so that we can extract information that isn't really part of the container settings, it is a global setting.

Reply:
I agree that it does make the `prepare_volume`'s signature worse.

However, `prepare_volume` is a SPRAS-internal API, and hash length is a setting passed to containers (as it is explicitly a dependency of `prepare_volume` used to set up the name of prepared volumes used in containers), making it a container setting.

The dependency diamond that motivates this change still only appears in #329 and not #390.